core/string/ustring.h

```diff
@@ -265,6 +265,9 @@ class String {
 	signed char nocasecmp_to(const String &p_str) const;
 	signed char naturalcasecmp_to(const String &p_str) const;
 	signed char naturalnocasecmp_to(const String &p_str) const;
+	// Special sorting for files to put files starting in `_` before all except `.`, and does natural comparison otherwise.
```

Comment:
Small nitpick :)
```suggestion
	// Special sorting for file names. Names starting with `_` are put before all others except `.`, otherwise natural comparison is used.
```
... or something like that. At least I'd replace `in` with `with`.